api/handler/attributes.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/nspcc-dev/neofs-s3-gw/api/errors"
+	"github.com/nspcc-dev/neofs-s3-gw/api/layer"
+)
+
+type (
+	GetObjectAttributesResponse struct {
+		ETag         string       `xml:"ETag,omitempty"`
+		ObjectSize   int64        `xml:"ObjectSize,omitempty"`
+		StorageClass string       `xml:"StorageClass,omitempty"`
+		ObjectParts  *ObjectParts `xml:"ObjectParts,omitempty"`
+	}
+
+	ObjectParts struct {
+		IsTruncated          bool   `xml:"IsTruncated,omitempty"`
+		MaxParts             int    `xml:"MaxParts,omitempty"`
+		NextPartNumberMarker int    `xml:"NextPartNumberMarker,omitempty"`
+		PartNumberMarker     int    `xml:"PartNumberMarker,omitempty"`
+		Parts                []Part `xml:"Part,omitempty"`
+		PartsCount           int    `xml:"PartsCount,omitempty"`
+	}
+
+	Part struct {
+		PartNumber int `xml:"PartNumber,omitempty"`
+		Size       int `xml:"Size,omitempty"`
+	}
+
+	GetObjectAttributesArgs struct {
+		MaxParts         int
+		PartNumberMarker int
+		Attributes       []string
+		VersionID        string
+	}
+)
+
+const (
+	maxPartsDefault         = 1000 // limit number of parts on a response
+	partNumberMarkerDefault = -1
+
+	eTag         = "ETag"
+	checksum     = "Checksum"
+	objectParts  = "ObjectParts"
+	storageClass = "StorageClass"
+	objectSize   = "ObjectSize"
+)
+
+var validAttributes = map[string]struct{}{
+	eTag:         {},
+	checksum:     {},
+	objectParts:  {},
+	storageClass: {},
+	objectSize:   {},
+}
+
+func (h *handler) GetObjectAttributesHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err  error
+		info *data.ObjectInfo
+
+		reqInfo = api.GetReqInfo(r.Context())
+	)
+
+	params, err := parseGetObjectAttributeArgs(r)
+	if err != nil {
+		h.logAndSendError(w, "invalid request", reqInfo, err)
+		return
+	}
+
+	bktInfo, err := h.getBucketAndCheckOwner(r, reqInfo.BucketName)
+	if err != nil {
+		h.logAndSendError(w, "could not get bucket info", reqInfo, err)
+		return
+	}
+
+	p := &layer.HeadObjectParams{
+		BktInfo:   bktInfo,
+		Object:    reqInfo.ObjectName,
+		VersionID: params.VersionID,
+	}
+
+	if info, err = h.obj.GetObjectInfo(r.Context(), p); err != nil {
+		h.logAndSendError(w, "could not fetch object info", reqInfo, err)
+		return
+	}
+
+	response, err := encodeToObjectAttributesResponse(info, params)
+	if err != nil {
+		h.logAndSendError(w, "couldn't encode object info to response", reqInfo, err)
+		return
+	}
+
+	writeAttributesHeaders(w.Header(), info, params)
+	if err = api.EncodeToResponse(w, response); err != nil {
+		h.logAndSendError(w, "something went wrong", reqInfo, err)
+	}
+}
+
+func writeAttributesHeaders(h http.Header, info *data.ObjectInfo, params *GetObjectAttributesArgs) {
+	h.Set(api.LastModified, info.Created.UTC().Format(http.TimeFormat))
+	if len(params.VersionID) != 0 {
+		h.Set(api.AmzVersionID, info.Version())
+	}
+
+	if _, ok := info.Headers[layer.VersionsDeleteMarkAttr]; ok {
+		h.Set(api.AmzDeleteMarker, strconv.FormatBool(true))
+	}
+
+	// x-amz-request-charged
```

Comment:
The header was skipped because I'm not sure what I should place in the header's value.